clang/include/clang/Sema/Sema.h

```diff
@@ -13059,9 +13059,8 @@ class Sema final {
 };
 
 template <typename AttrType>
-void Sema::addIntelSYCLSingleArgFunctionAttr(Decl *D,
-                                             const AttributeCommonInfo &CI,
-                                             Expr *E) {
+void Sema::addIntelSingleArgFunctionAttr(Decl *D, const AttributeCommonInfo &CI,
```

Comment:
I renamed this to handle both IntelSYCL and IntelFPGA attributes in the same function. I In my opinion this function needs refactoring but I didn't do it in this patch because I think this overlaps with what @AaronBallman is working on.

Reply:
LGTM. One nit

Since you are adding memory attribute inside addIntelSingleArgFunctionAttr(), so it would be better if you remove "Function" from the function name here to handle both function and memory attributes.

Reply:
Thanks for catching that! Done